pkg/apis/triggers/v1alpha1/trigger_template_validation.go

```diff
@@ -85,22 +85,30 @@ func validateResourceTemplates(templates []TriggerResourceTemplate) *apis.FieldE
 }
 
 // Verify every param in the ResourceTemplates is declared with a ParamSpec
-func verifyParamDeclarations(params []ParamSpec, templates []TriggerResourceTemplate) *apis.FieldError {
+func verifyParamDeclarations(ttParams []ParamSpec, templates []TriggerResourceTemplate) *apis.FieldError {
 	declaredParamNames := map[string]struct{}{}
-	for _, param := range params {
+	for _, param := range ttParams {
 		declaredParamNames[param.Name] = struct{}{}
 	}
 	for i, template := range templates {
-		// Get all params in the template $(params.NAME)
+		// Get all params in the template $(ttparams.NAME) or $(params.NAME)
 		templateParams := paramsRegexp.FindAllSubmatch(template.RawExtension.Raw, -1)
 		for _, templateParam := range templateParams {
-			templateParamName := string(templateParam[1])
+			templateParamName := string(templateParam[2])
 			if _, ok := declaredParamNames[templateParamName]; !ok {
+				// This logic is to get the tag and display in error dynamically for both ttparams and params.
+				// TODO(savitaashture): remove params once after the complete deprecation of spec.DeprecatedParams
+				var tag string
```

Comment:
Question: Until we start using only $(ttparams) for replacement, we cannot support https://github.com/tektoncd/triggers/issues/525 right?

Reply:
Yes because till that time we will have validation for `params`